tendermint/src/lite/verifier.rs

```diff
@@ -333,75 +337,81 @@ where
 // not store states twice.
 // Additionally, a new state is returned for convenience s.t. it can
 // be used for the other half of the recursion.
-fn verify_bisection_inner<H, C, L, R>(
-    trusted_state: &TrustedState<C, H>,
+fn verify_bisection_inner<'a, H, C, L, R>(
+    trusted_state: &'a TrustedState<C, H>,
     untrusted_height: Height,
     trust_threshold: L,
-    req: &R,
-    mut cache: &mut Vec<TrustedState<C, H>>,
-) -> Result<TrustedState<C, H>, Error>
+    req: &'a R,
+    mut cache: &'a mut Vec<TrustedState<C, H>>,
+) -> LocalBoxFuture<'a, Result<TrustedState<C, H>, Error>>
```

Comment:
It is a bit unfortunate that the we had to change the method signature to return a boxed future. But this change is good in the sense that it surfaces that there is still a too tight coupling between verification logic and the requester (or networking / IO).

@brapse also suggested yesterday that we should probably implement the iterative version here too. It seems like this is a good idea for several reasons now.

Reply:
https://github.com/interchainio/tendermint-rs/issues/174